taverna-scufl2-api/src/test/java/org/apache/taverna/scufl2/api/common/TestScufl2Tools.java

```diff
@@ -63,6 +64,25 @@ public void makeBundle() {
 		makeWorkflowBundle();
 		assertNotNull(workflowBundle);
 	}
+
+	@Test
+	public void testNestedWorkflows() {
+		Workflow child = new Workflow();
+		child.setName("childWorkflow");
+		child.setParent(workflowBundle);
+
+		Workflow mainWorkflow = workflowBundle.getMainWorkflow();
+		Processor processor = new Processor();
+		processor.setParent(mainWorkflow);
+
+		Profile profile = workflowBundle.getMainProfile();
+
+		Scufl2Tools tools = new Scufl2Tools();
+		tools.createNestedRelationship(processor, child, profile);
+		Workflow nested = tools.nestedWorkflowForProcessor(processor, profile);
+
+		assertEquals(child, nested);
```

Comment:
Could this test also verify that the corresponding `Activity` and `Configuration` have been added with the correct workflow name and i/o ports?

Reply:
I tried to add a patch where I check that the activity and configuration has been added to the workflow. But I'm not sure if it's what you asked for. Correct me please if you have something else in mind.

Reply:
Yeah, safely you should pick the correct activity based on the `ProcessorBinding` rather than picking blindly the first one with `NESTED_WORKFLOW` in it - although that will be the case for this particular test case profile as there should be no other activites.

So perhaps improve it to look up the ProcessorBinding for the processor and from there find the activity and its configuration? Sorry it's a bit cumbersome..

Reply:
I wasn't happy with that way either. I used the ProcessorBinding as you suggested and it's much better. Thank you for your comments !

Reply:
Great!